### ISSUE TITLE
Service messages get corrupted on SSH targets 

### ISSUE THREAD

**Issue by droyad** [OctopusDeploy/Issues#6731]
Partial service messages (i.e. `##octopus[`) are sometimes printed to the task log. It looks random. A [customer has reported](https://octopus.zendesk.com/agent/tickets/63324) this and we have [observed](https://octopusdeploy.slack.com/archives/C9TUPEJ8K/p1606189918198800?thread_ts=1605589456.132200&cid=C9TUPEJ8K) it in our integration tests. (Internal links)

A warning `Could not parse ##octopus[vcmVAUzE...` is displayed in the task logs. Note the lack of leading property and quotes. It looks like an extra `##octopus[` is getting inserted for some reason.

If this occurs on package acquisition, the following error may be displayed in the task logs:
```
Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')
17:18:47   Verbose  |       at System.Collections.Generic.List`1.get_Item(Int32 index)
                    |       at Octopus.Server.Orchestration.Targets.Common.PackageStagingMediator.TryFindPackageInCache(StoredPackage packageToPush, PackageAcquisitionExecutionHandlers execs, StagedPackage& stagedPackage) in 
```

## Workaround
Re-run the deployment

## Affected versions

Unsure at this stage, at least `2020.5.4` onwards.
Customer reported it starting after upgrading from `2019.13.5` to `2020.5.4`

**Comment by droyad** [comment]
I've been able to reproduce this issue. 1 in 1,500 deployments or so.

We are receiving the service message split over two lines. This is likely because the target is under heavy load and it's furiously trying to output messages over many SSH connections. When Windows outputs to a console, it usually manages to write a whole line at a time, but when under really heavy load, it'll start writing partial lines. I think Linux is doing the same thing here. Another possibility is that the messages are getting broken up in the TCP packets or in the SSH client's buffer. Again usually whole lines get sent, but under load it may take some time to write the whole thing. 

Now, looking at the tests, we do expect this to happen and handle this case. However for SSH targets we send a "I've got all the logs I can for now" after every line! This means the parser doesn't connect the two lines.

I think we can fix this by batching reads until the SSH client says "My read buffer is empty".
